connectors/vault/vaultconn-bl/vault_reader_test.go

```diff
@@ -0,0 +1,92 @@
+package vaultconnbl
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/http"
+	"github.com/hashicorp/vault/vault"
+
+	tu "github.com/ibm/the-mesh-for-data/connectors/vault/testutil"
+	vltutils "github.com/ibm/the-mesh-for-data/connectors/vault/vault_utils"
+	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
+)
+
+// TestVaultConnectorNormalRun tests the execution of GetCredentialsInfo with dummy mock id and
+// dummy credentials stored in the runtime test instance of vault
+func TestVaultConnectorNormalRun(t *testing.T) {
+	ln, client := createTestVault(t)
+	defer ln.Close()
+
+	srv := &Server{}
+	appID := "mock-appID"
+	//datasetID := "{\"ServerName\":\"cocoMDS3\",\"AssetGuid\":\"f6d9bf8c-dd37-4747-bca9-8ca9d1a5bb8f\"}"
+	datasetID := "mock-datasetID"
+	objToSendForCredential := &pb.DatasetCredentialsRequest{AppId: appID, DatasetId: datasetID}
+
+	config := vltutils.VaultConfig{
+		Token:   "token",
+		Address: "address",
+	}
+	connection := vltutils.VaultConnection{
+		Config: config,
+	}
+
+	data := map[string]interface{}{
+		"dataset_id":  datasetID,
+		"credentials": "my_egeria_credentials_test",
+	}
+
+	_, err := client.Logical().Write("secret/"+datasetID, data)
+	if err != nil {
+		t.Errorf("Error writing credentials from vault for " + datasetID + ":" + err.Error())
+	}
+
+	connection.Client = client
+	userVaultPath := tu.GetEnvironment()
+	var vault vltutils.VaultConnection
+	vault = connection
+	credentialsInfo, _ := srv.GetCredentialsInfo(objToSendForCredential, vault, userVaultPath)
+	log.Println("credentialsInfo")
```

Comment:
In case of a test failure `EnsureDeepEqualCredentials` will already log the credentials as it is part of the assert message. Thus I don't think you need to log them here separately.

Reply:
fixed. Thanks.